docs/blog/posts/responsive-app-background-task.md

```diff
@@ -0,0 +1,130 @@
+This is the code of the app so you can try it yourself:
+
+```py hl_lines="11-13 21 35 36"
+--8<-- "docs/blog/snippets/2022-12-07-responsive-app-background-task/blocking01.py"
+```
+
+1. The widget `ColourChanger` changes colours, randomly, when clicked.
+2. We create a binding to the key `l` that runs an action that we know will take some time (for example, reading and parsing a huge file).
+3. The method `action_load` is responsible for starting our time-consuming task and then reporting back.
+4. To simplify things a bit, our “time-consuming task” is just standing still for 5 seconds.
+
+I think it is easy to understand why the widget `ColourChanger` stops working when we hit the `time.sleep` call if we consider [the cooking analogy](https://mathspp.com/blog/til/cooking-with-asyncio) I have written about before in my blog.
+In short, Python behaves like a lone cook in a kitchen:
+
+ - the cook can be clever and multitask. For example, while water is heating up and being brought to a boil, the cook can go ahead and chop some vegetables.
+ - however, there is _only one_ cook in the kitchen, so if the cook is chopping up vegetables, they can't be seasoning a salad.
+
+Things like “chopping up vegetables” and “seasoning a salad” are _blocking_, i.e., they need the cook's time and attention.
+In the app that I showed above, the call to `time.sleep` is blocking, so the cook can't go and do anything else until the time interval elapses.
+
+## How can a cook multitask?
+
+It makes a lot of sense to think that a cook would multitask in their kitchen, but Python isn't like a smart cook.
+Python is like a very dumb cook who only ever does one thing at a time and waits until each thing is completely done before doing the next thing.
+So, by default, Python would act like a cook who fills up a pan with water, starts heating the water, and then stands there staring at the water until it starts boiling instead of doing something else.
+It is by using the module `asyncio` from the standard library that our cook learns to do other tasks while _awaiting_ the completion of the things they already started doing.
+
+[Textual](https://github.com/textualize/textual) is an async framework, which means it knows how to interoperate with the module `asyncio` and this will be the solution to our problem.
+By using `asyncio` with the tasks we want to run in the background, we will let the application remain responsive while we load and parse the data we need, or while we crunch the numbers we need to crunch, or while we connect to some slow API over the Internet, or whatever it is you want to do.
+
+The module `asyncio` introduces a keyword `async` that you can use to identify functions that you want to be able to run asynchronously.
+In other words, you use the keyword `async` to identify functions that contain tasks that would otherwise force the cook to waste time.
+(Functions with the keyword `async` are called _coroutines_.)
```

Comment:
Nitpick: The `async` keyword isn't introduced via the asyncio module, `async` is a language-level feature (you can have async frameworks other than asyncio that still use `async`)